homeassistant/components/ezviz/config_flow.py

```diff
@@ -0,0 +1,126 @@
+"""Config flow for ezviz."""
+import logging
+from typing import Any, Dict, Optional
+
+from pyezviz.client import EzvizClient, PyEzvizError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_REGION, CONF_TIMEOUT, CONF_USERNAME
+from homeassistant.core import callback
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .const import (
+    CONF_FFMPEG_ARGUMENTS,
+    DEFAULT_FFMPEG_ARGUMENTS,
+    DEFAULT_REGION,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def validate_input(hass: HomeAssistantType, data: dict) -> Dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    # constructor does login call
+    EzvizClient(
+        data[CONF_USERNAME],
+        data[CONF_PASSWORD],
+        data[CONF_REGION],
+        data.get(CONF_TIMEOUT, DEFAULT_TIMEOUT),
+    )
+
+
+@config_entries.HANDLERS.register(DOMAIN)
```

Comment:
The decorator is legacy. Please just pass the domain name as domain keyword argument to the class.

Reply:
Done.